blocks/basic/include/gnuradio-4.0/basic/FunctionGenerator.hpp

```diff
@@ -0,0 +1,297 @@
+#ifndef GNURADIO_FUNCTION_GENERATOR_HPP
+#define GNURADIO_FUNCTION_GENERATOR_HPP
+
+#include <gnuradio-4.0/Block.hpp>
+#include <gnuradio-4.0/Tag.hpp>
+#include <magic_enum.hpp>
+#include <magic_enum_utility.hpp>
+#include <numbers>
+
+namespace gr::basic {
+
+using namespace gr;
+
+namespace function_generator {
+enum class Type : int { Const, LinearRamp, ParabolicRamp, CubicSpline, ImpulseResponse };
+using enum Type;
+constexpr auto                                 TypeList  = magic_enum::enum_values<Type>();
+inline static constexpr gr::meta::fixed_string TypeNames = "[Const, LinearRamp, ParabolicRamp, CubicSpline, ImpulseResponse]";
+
+constexpr Type
+parse(std::string_view name) {
+    auto signalType = magic_enum::enum_cast<Type>(name, magic_enum::case_insensitive);
+    if (!signalType.has_value()) {
+        throw std::invalid_argument(fmt::format("unknown function generator type '{}'", name));
+    }
+    return signalType.value();
+}
+
+template<typename T>
+[[nodiscard]] property_map
+createConstPropertyMap(T startValue) {
+    return { { "signal_type", "Const" }, { "start_value", startValue } };
```

Comment:
minor:  maybe derive the "Const" and other magic string constants directly from the enum using `magic_enum::enum_name(...)` This would minimise future refactoring/renaming issues and also indicate to the user/other developers that these are API definitions.

Similarly `signal_type`, `start_value`, ... etc. could be also defined via an enum.